src/actions.js

```diff
@@ -29,7 +29,13 @@ export function fetchInstances() {
     }
 
     axios.get('https://instances.social/api/1.0/instances/list', { headers, params })
-         .then(({ data }) => dispatch(fetchInstancesSuccess(data.instances)))
+         .then(({ data }) => {
+            if (data.instances.length == 0) {
+              throw new Error("No data returned");
+            }
+
+            return dispatch(fetchInstancesSuccess(data.instances));
```

Comment:
I think there needs to be an `axios.get` to `/cache` here, like below

Reply:
The throw moves execution to the catch block.

Reply:
Oops, right.